src/validation/h5p.ts

```diff
@@ -0,0 +1,308 @@
+/**
+ * This namespace contains TypeScript definitions for the H5P specification
+ * as described at https://h5p.org/developers
+ */
+export namespace H5P {
+  /**
+   * File extension of H5P packages (not dot-prefixed)
+   */
+  export const H5P_FILE_EXTENSION = 'h5p';
+
+  /**
+   * File extensions that are allowed inside an H5P package
+   */
+  export const ALLOWED_FILE_EXTENSIONS = [
+    'bmp',
+    'css',
+    'csv',
+    'diff',
+    'doc',
+    'docx',
+    'eof',
+    'gif',
+    'jpeg',
+    'jpg',
+    'js',
+    'json',
+    'mp3',
+    'mp4',
+    'm4a',
+    'odp',
+    'ods',
+    'odt',
+    'ogg',
+    'otf',
+    'patch',
+    'pdf',
+    'png',
+    'ppt',
+    'pptx',
+    'rtf',
+    'svg',
+    'swf',
+    'textile',
+    'tif',
+    'tiff',
+    'ttf',
+    'txt',
+    'wav',
+    'webm',
+    'woff',
+    'xls',
+    'xlsx',
+    'xml',
+    'md',
+    'textile',
+    'vtt',
+  ];
+
+  /**
+   * This interface represents a valid h5p.json manifest
+   * The specification is found at
+   * https://h5p.org/documentation/developers/json-file-definitions
+   */
+  export interface Manifest {
+    /* Mandatory properties */
+
+    /** The title of the H5P would typically be used on pages displaying it, and in system administration lists. This can be any valid string. */
+    title: string;
+    /** The main H5P library for this content. This library will be initialized with the content data from the content folder. */
+    mainLibrary: string;
+    /** A standard language code. We are using the ISO-639-1 to classify all supported languages. This is a two-letter code, for example 'en' for English. A list of ISO-639-1 codes can be found at https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes. Set "und" for language neutral content. */
+    language:
```

Comment:
This is a copy of what's in the schema, right? Any way of factoring it out?

Reply:
This is the type definition which the schema validates against. This cannot be factored out for the same reason as in https://github.com/graasp/graasp-plugin-h5p/pull/5#discussion_r921985782